substrate/frame/examples/single-block-migrations/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Single Block Migration Example Pallet
+//!
+//! An example pallet demonstrating best-practices for writing single-block migrations in the
+//! context of upgrading pallet storage.
+//!
+//! Read more about runtime upgrades and migrations by reading the [Runtime Upgrade Reference
+//! Docs](runtime_upgrade_reference_docs).
+//!
+//! ## Forwarning
+//!
+//! Single block migrations **MUST** execute in a single block, therefore are typically only
+//! appropriate for migrations which are either guaranteed to not exceed block weight limits.
```

Comment:
There some part of the sentence missing. In general its only an issue for parachains.